### PR TITLE
Update Node and NPM to lts/gallium

### PR DESCRIPTION
**Pull request by daniellacosse**
Server PR: https://github.com/Jigsaw-Code/outline-server/pull/1004

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -2,9 +2,10 @@
   "name": "outline-client",
   "productName": "Outline",
   "scripts": {
+    "action": "bash ./scripts/run_action.sh",
+    "action:list": "find . -name '*.action.sh' | sed -E 's:./(.*).action.sh:\\1:' | grep -v 'scripts/run_action'",
     "clean": "rm -rf build www node_modules platforms/* plugins/*",
-    "do": "bash ./scripts/do_action.sh",
-    "actions": "find . -name '*_action.sh' | sed -E 's:./src/(.*)_action.sh:\\1:' | grep -v 'scripts/do_action'"
+    "lint:ts": "tslint --fix 'src/**/*.ts'"
```
Comment (fortuna) [inline package.json:8]: Having a lint command change code is unexpected. Can you remove the `--fix`?

File: package.json [hunk 2]
```diff
@@ -109,7 +110,7 @@
   "main": "build/electron/electron/index.js",
   "husky": {
     "hooks": {
-      "pre-commit": "npx tslint --fix 'src/**/*.ts' && npx git-clang-format && npx pretty-quick --staged --pattern '**/*.html'"
+      "pre-commit": "npm run lint:ts && npx git-clang-format && npx pretty-quick --staged --pattern '**/*.html'"
```
Comment (fortuna) [inline package.json:113]: Add the --fix here.

I'm not a big fan of this pre-commit by the way. We may want to rethink it (not for this PR)

File: package.json (hunk 1 above)
Comment (daniellacosse) [reply package.json:8]: Ah, right

File: package.json (hunk 2 above)
Comment (daniellacosse) [reply package.json]: Done.

Totally agree, I really dislike commit hooks myself. Not only can they just be bypassed, I feel dev velocity is impacted by:

- lack of live-reloading
- lack of supported editor (debatable) 
- lack of consistent folder structure, templating, and snippets
- coordinating updates across repositories
- lack of automated testing to debug against (this includes storybook)
- slow CI
- lack of fully automated deployments

**Review by fortuna — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -7,9 +7,14 @@ The client's user interface is implemented in [Polymer](https://www.polymer-proj
 
 ## Requirements for all builds
 
-All builds require [npm](https://docs.npmjs.com/downloading-and-installing-node-js-and-npm), in addition to other per-platform requirements. After cloning this repo install all dependencies with the command.
+All builds require the latest TLS of [Node](https://nodejs.org/), in addition to other per-platform requirements.
```
Comment (bemasc) [inline README.md:10]:
```suggestion
All builds require [Node](https://nodejs.org/) 16, in addition to other per-platform requirements. 
```

File: package.json [hunk 4]
```diff
@@ -2,9 +2,11 @@
   "name": "outline-client",
   "productName": "Outline",
   "scripts": {
+    "action": "bash ./scripts/run_action.sh",
+    "action:help": "npm run action",
+    "action:list": "npm run action",
     "clean": "rm -rf build www node_modules platforms/* plugins/*",
-    "do": "bash ./scripts/do_action.sh",
-    "actions": "find . -name '*_action.sh' | sed -E 's:./src/(.*)_action.sh:\\1:' | grep -v 'scripts/do_action'"
+    "lint:ts": "tslint 'src/**/*.ts'"
```
Comment (bemasc) [inline package.json:9]: I think we should have a `"lint"` action.  If you want, it can be an alias to `lint:ts`.

File: package.json [hunk 5]
```diff
@@ -109,7 +111,7 @@
   "main": "build/electron/electron/index.js",
   "husky": {
     "hooks": {
-      "pre-commit": "npx tslint --fix 'src/**/*.ts' && npx git-clang-format && npx pretty-quick --staged --pattern '**/*.html'"
+      "pre-commit": "npm run lint:ts -- --fix && npx git-clang-format && npx pretty-quick --staged --pattern '**/*.html'"
```
Comment (bemasc) [inline package.json:114]: Can we make `clang-format` or `pretty-quick` into lint actions that check only changed lines?

File: scripts/run_action.sh [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+  local -r STYLE_RESET='\033[0m'
+
+  local -r action="${1:-""}"
+  local -r old_indent="${run_action_indent}"
+
+  run_action_indent="=> ${run_action_indent}"
+
+  if [[ -z "${action}" ]]; then
+    echo -e "Please provide an action to run. ${STYLE_BOLD_WHITE}List of valid actions:${STYLE_RESET}\n"
+    find . -name '*.action.sh' | sed -E 's:./(.*)\.action\.sh:\1:'
```
Comment (bemasc) [inline scripts/run_action.sh:40]:
```suggestion
    find . -name '*.action.sh' | sed -E 's:\./(.*)\.action\.sh:\1:'
```

File: package.json (hunk 5 above)
Comment (fortuna) [reply package.json:114]: Maybe in a follow up PR? Let's get this one in first.

File: package.json (hunk 4 above)
Comment (daniellacosse) [reply package.json:9]: yes, love it

**Review by bemasc — APPROVED** [review]

**Pull request merged**